test/EndToEnd/GivenPortableApps.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using NuGet.ProjectModel;
+using NuGet.Versioning;
+using Xunit;
+
+namespace EndToEnd
+{
+    public class GivenPortableApps : TestBase
+    {
+        [Theory]
+        [ClassData(typeof(SupportedNetCoreAppVersions))]
+        public void ItDoesNotRollsForwardToTheLatestVersion(string minorVersion)
+        {
+            var _testInstance = TestAssets.Get("TestAppSimple")
+                .CreateInstance(identifier: minorVersion)
+                .WithSourceFiles();
+
+            string projectDirectory = _testInstance.Root.FullName;
+
+            string projectPath = Path.Combine(projectDirectory, "TestAppSimple.csproj");
+
+            var project = XDocument.Load(projectPath);
+            var ns = project.Root.Name.Namespace;
+
+            //  Update TargetFramework to the right version of .NET Core
+            project.Root.Element(ns + "PropertyGroup")
+                .Element(ns + "TargetFramework")
+                .Value = "netcoreapp" + minorVersion;
+
+            project.Save(projectPath);
+
+            //  Get the version rolled forward to
+            new RestoreCommand()
+                    .WithWorkingDirectory(projectDirectory)
+                    .Execute()
+                    .Should().Pass();
+
+            string assetsFilePath = Path.Combine(projectDirectory, "obj", "project.assets.json");
+            var assetsFile = new LockFileFormat().Read(assetsFilePath);
+
+            var versionInAssertsJson = GetNetCoreAppVersion(assetsFile);
+            versionInAssertsJson.Should().NotBeNull();
+
+            if (versionInAssertsJson.IsPrerelease)
```

Comment:
For this test, I think the condition here should be:

```c#
if (versionInAssertsJson.IsPrerelease && versionInAssetsJson.Patch == 0)
```

This ensures that if the bundled version is, for example, a prerelease of .NET Core 2.1.1, that we don't roll forward to that prerelease version for framework-dependent deployments.

Reply:
Fixed and updated the comment